modules/payment/src/Plugin/Commerce/CheckoutPane/PaymentProcess.php

```diff
@@ -133,38 +133,43 @@ public function buildPaneForm(array $pane_form, FormStateInterface $form_state,
     $payment_gateway = $this->order->payment_gateway->entity;
     $payment_gateway_plugin = $payment_gateway->getPlugin();
 
+    $payment_storage = $this->entityTypeManager->getStorage('commerce_payment');
+    $payment = $payment_storage->create([
```

Comment:
We don't save it. We use it so that the `offsite-payment` knows the order and any other contexts when building the form. I chose that to be the `EntityWithPaymentGatewayInterface` passed.